### DISCUSSION TITLE
Is there a tutorial or a README page for the terraform script?

### DISCUSSION THREAD

**Discussion by sonamsamdupkhangsar** [digitalocean/Kubernetes-Starter-Kit-Developers#202] [Q&A] [answered]
Hello DO team,

I saw the assets folder that has the terraform scripts.  Do you have a README page that describes what it does or how to setup a cluster using that script?

Thanks
-Sonam

**Comment by diabhey** [comment]
Hi @sonamsamdupkhangsar 
We use [terraform](https://www.terraform.io/), which is an IaC (Infrastructure as Code) tool to set up our DigitalOcean Kubernetes cluster. Perhaps you can follow the tutorial [here](https://github.com/digitalocean/production-ready-kubernetes-workshop/blob/main/00-cluster-setup.md) to get a better understanding.

**Reply by sonamsamdupkhangsar** [reply]
Thanks @diabhey That tutorial should help.

**Comment by v-ctiutiu** [comment, marked as answer]
@sonamsamdupkhangsar @diabhey 

The Terraform setup from the assets folder is not meant to be used as part of any tutorials from the Starter Kit. We started to write some automated tests at some point in time for the Starter Kit itself. We wanted to make sure that when upgrading Starter Kit components (and the associated Helm charts), some automatic validation takes place to make sure that nothing breaks. The [Terraform stuff](https://github.com/digitalocean/Kubernetes-Starter-Kit-Developers/tree/main/assets/terraform) is used in the [automated tests GitHub workflow ](https://github.com/digitalocean/Kubernetes-Starter-Kit-Developers/blob/main/.github/workflows/test.yml) associated with this repository.

If you're still curios what it does, I will summarize it for you:

1. A DOKS cluster is provisioned first via Terraform.
2. Then, after the cluster is up and running, the [Nginx ingress controller chart](https://github.com/kubernetes/ingress-nginx/tree/main/charts/ingress-nginx) is installed via Terraform using the [Helm provider](https://registry.terraform.io/providers/hashicorp/helm/latest/docs).
3. [Nginx installation is tested ](https://github.com/digitalocean/Kubernetes-Starter-Kit-Developers/blob/main/assets/kubetest/test_nginx_ingress.py) afterwards using [pytest](https://docs.pytest.org/) and [kubetest](https://kubetest.readthedocs.io/en/latest/).

**Note:**

Only the Nginx ingress controller was added for testing. We had plans to add the rest of the components used in the Starter Kit but it was dropped unfortunately.

@sonamsamdupkhangsar I think you also have a point, and maybe adding a README explaining why it's there would help other readers understand the scope.

I hope you have a clear overview and answer now about the Terraform stuff from the assets folder.

Thanks.

**Reply by sonamsamdupkhangsar** [reply]
Thanks @v-ctiutiu for the summary.
